src/sentry/static/sentry/app/views/eventsV2/eventView.tsx

```diff
@@ -377,23 +377,38 @@ class EventView {
     });
   }
 
-  static fromSavedQueryWithLocation(
-    saved: NewQuery | LegacySavedQuery,
-    location: Location
-  ): EventView {
+  static fromNewQueryWithLocation(newQuery: NewQuery, location: Location): EventView {
     const query = location.query;
 
-    saved = {
-      ...saved,
-      start: saved.start || decodeScalar(query.start),
-      end: saved.end || decodeScalar(query.end),
-      range: saved.range || decodeScalar(query.statsPeriod),
+    // apply global selection header values from location whenever possible
+
+    const environment: string[] =
+      Array.isArray(newQuery.environment) && newQuery.environment.length > 0
+        ? newQuery.environment
+        : collectQueryStringByKey(location.query, 'environment');
+
+    const project: number[] =
+      Array.isArray(newQuery.projects) && newQuery.projects.length > 0
+        ? newQuery.projects
+        : decodeProjects(location);
+
+    const saved: NewQuery = {
+      ...newQuery,
+
+      environment,
+      projects: project,
+
+      // datetime selection
+
+      start: newQuery.start || decodeScalar(query.start),
+      end: newQuery.end || decodeScalar(query.end),
+      range: newQuery.range || decodeScalar(query.statsPeriod),
     };
 
     return EventView.fromSavedQuery(saved);
   }
 
-  static fromSavedQuery(saved: NewQuery | LegacySavedQuery): EventView {
+  static fromSavedQuery(saved: NewQuery | SavedQuery): EventView {
```

Comment:
You could remove the `isLegacySavedQuery()` conditions too.

Reply:
Done.